wails-frontend/src/pages/importPrivateKey.tsx

```diff
@@ -1,97 +1,131 @@
-import { useState } from 'react';
+import { SyntheticEvent, useRef, useState } from 'react';
 import { promptRequest } from '../events/events';
 import { useLocation, useNavigate } from 'react-router-dom';
 import { handleCancel } from '../utils/utils';
 import { Password, Input, Button } from '@massalabs/react-ui-kit';
+import { IErrorObject, parseForm } from '../utils';
+import {
+  IsAlreadyExists,
+  IsNicknameValid,
+} from '../../wailsjs/go/walletapp/WalletApp';
+
+// TODO: create i18n and move this to translation file
+const t = (key: string): string => {
+  const errors: Record<string, string> = {
+    'nickname-required': 'Nickname is required',
+    'nickname-already-exists': 'This account name already exists',
+    'nickname-invalid-format':
+      "The account name can't contain any special characters",
+  };
+
+  return errors[key] || key;
+};
 
 const ImportPrivatekey = () => {
   const navigate = useNavigate();
+  const form = useRef(null);
 
   const [input, setInput] = useState<string | undefined>(undefined);
   const [privateKey, setPkey] = useState<string | undefined>(undefined);
+  const [error, setError] = useState<IErrorObject | null>(null);
 
   let { state } = useLocation();
   const req: promptRequest = state.req;
 
   const applyStr = 'Next';
 
-  const baselineStr = () => {
-    if (privateKey) {
-      return 'Choose an account name';
+  const baselineStr = () =>
+    privateKey ? 'Define your account name' : 'Enter your Private Key';
+
+  const placeholder = () => (privateKey ? 'Account name' : 'Private key');
+
+  async function validateNickname(e: SyntheticEvent) {
+    const formObject = parseForm(e);
+    const { nickname } = formObject;
+
+    if (!nickname) {
+      setError({ nickname: t('nickname-required') });
```

Comment:
Non blocking: If we could, as much as possible, assign the key as `page.sub.field-validation` it will be, IMHO, easy to manage when this became big. So, in this case something like 'import.nickname-required`.

Reply:
Ok! Good for me! Just need to keep track of that. Set some task including the i18N on placeholder. Otherwise we'll miss it! 😅 

Feel free to close here then...